src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
+        // 'true' or 'false', it will be used as is.  If the value is
+        // empty, and 'detectFunc' is specified, it will be invoked to
+        // evaluate the condition at runtime.  The value is cached, so
+        // 'detectFunc' will be called only once.
+        private static bool GetConditionValueFromTestProperties(string conditionName, Func<bool> detectFunc = null)
+        {
+            string propertyValue;
+            return GetConditionValue(conditionName, () =>
+            {
+                propertyValue = TestProperties.GetProperty(conditionName);
+                if (string.IsNullOrWhiteSpace(propertyValue))
+                {
+                    // Empty string means auto-detect if possible, else false
+                    return (detectFunc == null) ? false : detectFunc();
+                }
+                else
+                {
+                    bool conditionResult = false;
+                    if (!bool.TryParse(propertyValue, out conditionResult))
+                    {
+                        conditionResult = false;
+                    }
+
+                    return conditionResult;
+                }
+            });
+        }
+
+        private static bool Detect_Server_Is_Local()
+        {
+            // Temporary workaround to detect whether this test is using localhost
+            // for either new or old test infrastructure.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            string host = null;
+            if (TestProperties.PropertyNames.Contains("ServiceUri"))
+            {
+                host = TestProperties.GetProperty("ServiceUri");
+            }
+            else if (TestProperties.PropertyNames.Contains("BridgeHost"))
+            {
+                host = TestProperties.GetProperty("BridgeHost");
+            }
+             if (String.IsNullOrWhiteSpace(host))
+            {
+                return false;
+            }
+            int index = host.IndexOf("localhost", 0, StringComparison.OrdinalIgnoreCase);
+            return index >= 0;
+        }
+        
+        private static bool Detect_Is_Windows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
         }
 
         public static bool Domain_Joined()
         {
-            return IsConditionalTestEnabled(TestProperties.Domain_Joined_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // the client and server are domain joined.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Domain_Joined_PropertyName,
+                                                       Detect_Server_Is_Local);
```

Comment:
+1. This will be a better way to implement "is domain joined", but this can only detect if client machine is domain joined. Is "server domain joined" part of this condition or is it separate?